node_modules/col-whiteboards/lib/api.js

```diff
@@ -0,0 +1,771 @@
+/**
+ * Copyright 2015 UC Berkeley (UCB) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('lodash');
+var config = require('config');
+var cookie = require('cookie');
+var cookieParser = require('cookie-parser');
+var Joi = require('joi');
+var url = require('url');
+
+var Collabosphere = require('col-core');
+var CollabosphereUtil = require('col-core/lib/util');
+var DB = require('col-core/lib/db');
+var log = require('col-core/lib/logger')('col-whiteboards');
+var UserConstants = require('col-users/lib/constants');
+
+/**
+ * Get a whiteboard
+ * // TODO: Only load online people and whiteboard content when required
+ *
+ * @param  {Number}         userId                          The id of the user retrieving the whiteboard
+ * @param  {Number}         id                              The id of the whiteboard
+ * @param  {Function}       callback                        Standard callback function
+ * @param  {Object}         callback.err                    An error that occurred, if any
+ * @param  {Whiteboard}     callback.whiteboard             The requested whiteboard
+ */
+var getWhiteboard = module.exports.getWhiteboard = function(userId, id, callback) {
```

Comment:
We should pass in the request context here rather than the userId. That will allow us to add the course to the `where` parameters. As it stands, users from other courses could potentially request whiteboards they shouldn't be able to see